src/v1/device_manager_client.ts

```diff
@@ -2686,30 +2688,37 @@ export class DeviceManagerClient {
     });
     // eslint-disable-next-line no-async-promise-executor
     return new Promise<string>(async (resolve, reject) => {
-      const options = {
-        host: this.BASE_URL,
-        path:
-          `/api/v/1/code/` + this.ADMIN_SYSTEM_KEY + `/getRegistryCredentials`,
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-          'ClearBlade-UserToken': this.ADMIN_USER_TOKEN,
-          'Content-Length': payload.length,
-        },
-      };
-      const req = https.request(options, res => {
-        let data = '';
-        const chunks: any[] = [];
-        res.on('data', chunk => (data += chunk));
-        res.on('end', () => {
-          resolve(data);
+      if (this.CACHED_CLIENT_INFO) {
```

Comment:
Yes, in current implementation I have stored whole JSON response, so directly parsing it from there. If you think that's not descent way then I can store it in Map instead of current one.